### PR TITLE
node checks: verify node 5

### PR DESCRIPTION
**Pull request by nathanleiby**
- checks that a supported base image is used in Dockerfile
- checks that `nvm install <version>` exists in .drone.yml and installs
  the correct version

### CODE REVIEW

**Comment by nathanleiby** [comment]
nathan's wishlist: I think with a small-medium amount of further report-card work we could add a datastore, SQL query it, and produce a spreadsheet like the below automatically (
https://clever.atlassian.net/wiki/display/ENG/Node+Versions)

**Comment by nathanleiby** [comment]
cc: @ericavonb @jefff @cozmo (node guild)

File: report_card/checks/node_checks.py [hunk 1]
```diff
@@ -83,3 +89,37 @@ def test_unused_packages(self):
...
         self.assertEqual(len(unused), 0,
                          'unused dependencies in package.json:\n{}'.format(' '.join(unused)))
+
+    def test_dockerfile_node_base_image(self):
+        """ Dockerfile must use a supported Node base image
+        """
+        unittest.skipUnless(os.path.isfile(dockerfile_loc), 'No Dockerfile in this repo')
+
+        whitelist = [
+            'node:5.4.1-slim'
```
Comment (peggyl) [inline report_card/checks/node_checks.py]: Curious: why do we only allow 5.4.1-slim? The latest is 5.7.0: https://hub.docker.com/r/library/node/tags/

File: report_card/checks/node_checks.py [hunk 2]
```diff
@@ -83,3 +89,37 @@ def test_unused_packages(self):
...
+        ]
+
+        dfp=DockerfileParser(repo)
```
Comment (peggyl) [inline report_card/checks/node_checks.py]: nit: spaces around `=` (maybe not worth fixing unless other changes needed too)

**Comment by peggyl** [comment]
:a: overall. Would like to know why we're standardizing on node 5.4.1 and not 5.7.0 (latest) or any version of 5 (`nvm install 5` installs 5.7.0) - didn't see 5.4.1 explicitly mentioned in any emails or Slack channels, but sorry if I missed this!

**Comment by peggyl** [comment]
:gift: re: wishlist - could probably do something similar with a gitbot-like script (use Github library/API to get node versions in all our node repos and do something with them, like generate a report)!

File: report_card/checks/node_checks.py [hunk 3]
```diff
@@ -83,3 +89,37 @@ def test_unused_packages(self):
...
+          if d['instruction'] == 'FROM':
+              image = d['value']
+              ok = any(map(lambda w: image == w, whitelist))
+              self.assertTrue(ok, "using unsupported node image: {}".format(image))
+
+    def test_drone_node_version(self):
+        """ .drone.yml installs Node v5.4.1
+        """
+        drone_yml = _drone_yml_or_skip()
+        expected_version = 'v5.4.1'
```
Comment (jefff) [inline report_card/checks/node_checks.py]: Does it make sense to just check for Node 5 and not worry about minor version?

File: report_card/checks/node_checks.py (hunk 1 above)
Comment (cozmo) [inline report_card/checks/node_checks.py]: We also use later versions - see for example https://github.com/Clever/launchpad/blob/master/Dockerfile#L1

Comment (rgarcia) [inline report_card/checks/node_checks.py]: also https://github.com/Clever/school-admin-dashboard/blob/master/Dockerfile#L1

Comment (nathanleiby) [inline report_card/checks/node_checks.py]: I left it as a whitelist to be more flexible as we got feedback. 

welp, I thought I was using the version from one of our repos but now I can't find it. I'll update to 5.7.0

Comment (nathanleiby) [inline report_card/checks/node_checks.py]: It was this one: gearman-admin https://github.com/Clever/gearman-admin/pull/

Comment (nathanleiby) [inline report_card/checks/node_checks.py]: @cozmo should we support 5.x or try to get everything moved to 5.7?

File: report_card/checks/node_checks.py (hunk 3 above)
Comment (nathanleiby) [inline report_card/checks/node_checks.py]: @jefff yeah, probably. I asked Cozmo what he thinks as well

File: report_card/checks/node_checks.py (hunk 2 above)
Comment (nathanleiby) [inline report_card/checks/node_checks.py]: @peggyl I'll come back after this PR and do an automated formatting step

File: report_card/checks/node_checks.py (hunk 3 above)
Comment (nathanleiby) [inline report_card/checks/node_checks.py]: Made checks generic to 5.x  d13cdfc

File: report_card/checks/node_checks.py (hunk 1 above)
Comment (nathanleiby) [inline report_card/checks/node_checks.py]: Made checks generic to 5.x  d13cdfc

**Comment by peggyl** [comment]
LGTM. Thanks for allowing 5.x! :+1:

**Pull request merged**